### PR TITLE
Add Llama3 tokenizer and allow custom tokenizers.

### PR DESCRIPTION
**Pull request by Priva28**
This PR:
- Adds a tokenizer for llama3. 
   - The tokenizer is used if `"llama3"` is found in the model string.
   - https://huggingface.co/Xenova/llama-3-tokenizer/tree/main
- Allows using a custom tokenizer so users don't have to wait for it to be added to the repo.
  - A custom tokenizer can be created with `create_pretrained_tokenizer` and `create_tokenizer`.
  - These methods behave the same as `Tokenizer.from_pretrained` and `Tokenizer.from_str` but return the result in a dict format that can be used easily with the `token_counter` method.
  - Returns in this format (same as `_select_tokenizer`): `{"type": "huggingface_tokenizer", "tokenizer": Tokenizer}`

### CODE REVIEW

**Comment by ghost** [comment]
hey @Priva28 can you add instructions on how to do this to docs - https://docs.litellm.ai/docs/completion/token_usage

**Comment by Priva28** [comment]
Hi @krrishdholakia, I added some instructions to the docs and also allowed passing the custom_tokenizer to the encode and decode helpers. Lmk if this looks good, maybe there's a more clear way to describe its purpose?

File: docs/my-website/docs/completion/token_usage.md [hunk 1]
```diff
@@ -60,7 +62,24 @@ messages = [{"user": "role", "content": "Hey, how's it going"}]
 print(token_counter(model="gpt-3.5-turbo", messages=messages))
 ```
 
-### 4. `cost_per_token`
+### 4. `create_pretrained_tokenizer` and `create_tokenizer`
+
+```python
```
Comment (ghost) [inline docs/my-website/docs/completion/token_usage.md:67]: this is great! 

Can we add this as a test here, and confirm it passes - https://github.com/BerriAI/litellm/blob/f7d7d3426191009f199d09e1aadd7be72ca94458/litellm/tests/test_token_counter.py#L52

To make sure we don't have future regressions @Priva28

**Comment by ghost** [comment]
Hi @Priva28 can you confirm this passes your testing? (screenshot works!)

**Comment by Priva28** [comment]
Tests passing! @krrishdholakia 
Screenshot 2024-05-02 at 4 04 32 pm

**Comment by ghost** [comment]
LGTM!

**Comment by ishaan-jaff** [comment]
hi @Priva28 curious do you use the litellm proxy in production today ?

**Pull request merged**